controllers/scc.go

```diff
@@ -0,0 +1,180 @@
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	secv1 "github.com/openshift/api/security/v1"
+	lvmv1alpha1 "github.com/red-hat-storage/lvm-operator/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	sccName = "topolvm-scc"
+)
+
+type openshiftSccs struct{}
+
+// openshiftSccs unit satisfies resourceManager interface
+var _ resourceManager = openshiftSccs{}
+
+func (c openshiftSccs) getName() string {
+	return sccName
+}
+
+func (c openshiftSccs) ensureCreated(r *LVMClusterReconciler, ctx context.Context, lvmCluster *lvmv1alpha1.LVMCluster) error {
+	if !IsOpenshift(r) {
+		r.Log.Info("not creating SCCs as this is not an Openshift cluster")
+		return nil
+	}
+	sccs := getAllSCCs(lvmCluster.Namespace)
+	for _, scc := range sccs {
+		_, err := r.SecurityClient.SecurityContextConstraints().Get(ctx, scc.Name, metav1.GetOptions{})
+		if err != nil && errors.IsNotFound(err) {
+			r.Log.Info("creating SecurityContextConstraint", "SecurityContextConstraint", scc.Name)
+			_, err := r.SecurityClient.SecurityContextConstraints().Create(ctx, scc, metav1.CreateOptions{})
+			if err != nil {
+				return fmt.Errorf("failed to create SCC %q: %v", scc.Name, err)
+			}
+		} else if err == nil {
+			//Don't update the SCC
+			r.Log.Info("already exists", "SecurityContextConstraint", scc.Name)
+		} else {
+			r.Log.Error(err, "Something went wrong when checking for SecurityContextConstraint", "SecurityContextConstraint", scc.Name)
+			return fmt.Errorf("something went wrong when checking for SCC %q: %v", scc.Name, err)
+		}
+	}
+
+	return nil
+}
+
+func (c openshiftSccs) ensureDeleted(r *LVMClusterReconciler, ctx context.Context, lvmCluster *lvmv1alpha1.LVMCluster) error {
+	if IsOpenshift(r) {
+		var err error
+		sccs := getAllSCCs(lvmCluster.Namespace)
+		for _, scc := range sccs {
+			err = r.SecurityClient.SecurityContextConstraints().Delete(ctx, scc.Name, metav1.DeleteOptions{})
+			if err != nil {
+				if errors.IsNotFound(err) {
+					r.Log.Info("SecurityContextConstraint is already deleted", "SecurityContextConstraint", scc.Name)
+					return nil
+				} else {
+					r.Log.Error(err, "failed to delete SecurityContextConstraint", "SecurityContextConstraint", scc.Name)
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func (c openshiftSccs) updateStatus(r *LVMClusterReconciler, ctx context.Context, lvmCluster *lvmv1alpha1.LVMCluster) error {
+	// intentionally empty
+	return nil
+}
+
+func getAllSCCs(namespace string) []*secv1.SecurityContextConstraints {
+	return []*secv1.SecurityContextConstraints{
+		newTopolvmNodeScc(namespace),
+		newVGManagerScc(namespace),
+	}
+}
+
+func newTopolvmNodeScc(namespace string) *secv1.SecurityContextConstraints {
+	scc := &secv1.SecurityContextConstraints{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "security.openshift.io/v1",
+			Kind:       "SecurityContextConstraints",
+		},
+	}
+	scc.Name = "odf-lvm-topolvm-node"
```

Comment:
is the prefix `odf` intentional here?

Reply:
Yes. SCCs will be created only on Openshift so Odf is fine.